scripts/mitochondria_m2_wdl/AlignAndCall.wdl

```diff
@@ -335,3 +355,172 @@ task LiftoverAndCombineVcfs {
         File final_vcf_index = "${basename}.final.vcf.idx"
     }
 }
+
+task M2 {
+  File ref_fasta
+  File ref_fai
+  File ref_dict
+  File input_bam
+  File input_bai
+  String? m2_extra_args
+  Boolean? make_bamout
+  Boolean compress
+  File? gga_vcf
+  File? gga_vcf_idx
+
+  String output_vcf = "raw" + if compress then ".vcf.gz" else ".vcf"
+  String output_vcf_index = output_vcf + if compress then ".tbi" else ".idx"
+
+  File? gatk_override
+
+  # runtime
+  Int? mem
+  Int? preemptible_tries
+  Float ref_size = size(ref_fasta, "GB") + size(ref_fai, "GB")
+  Int disk_size = ceil(size(input_bam, "GB") + ref_size) + 20
+
+  # Mem is in units of GB but our command and memory runtime values are in MB
+  Int machine_mem = if defined(mem) then mem * 1000 else 3500
+  Int command_mem = machine_mem - 500
+
+  meta {
+    description: "Mutect2 for calling Snps and Indels"
+  }
+  parameter_meta {
+    input_bam: "Aligned Bam"
+    gga_vcf: "VCF for genotype given alleles mode"
+  }
+  command <<<
+      set -e
+
+      export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk_override}
+
+      # We need to create these files regardless, even if they stay empty
+      touch bamout.bam
+
+      gatk --java-options "-Xmx${command_mem}m" Mutect2 \
+        -R ${ref_fasta} \
+        -I ${input_bam} \
+        ${"--genotyping-mode GENOTYPE_GIVEN_ALLELES --alleles " + gga_vcf} \
+        -O ${output_vcf} \
+        ${true='--bam-output bamout.bam' false='' make_bamout} \
+        ${m2_extra_args} \
+        --annotation StrandBiasBySample \
+        --mitochondria-mode \
+        --max-reads-per-alignment-start 75 \
+        --max-mnp-distance 0
+  >>>
+  runtime {
+      docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+      memory: machine_mem + " MB"
+      disks: "local-disk " + disk_size + " HDD"
+      preemptible: select_first([preemptible_tries, 5])
+      cpu: 2
+  }
+  output {
+      File raw_vcf = "${output_vcf}"
+      File faw_vcf_idx = "${output_vcf_index}"
+      File stats = "${output_vcf}.stats"
+      File output_bamOut = "bamout.bam"
+  }
+}
+
+task Filter {
+  File ref_fasta
+  File ref_fai
+  File ref_dict
+  File raw_vcf
+  File raw_vcf_index
+  File raw_vcf_stats
+  Boolean compress
+  Float? vaf_cutoff
+
+  String output_vcf = "output" + if compress then ".vcf.gz" else ".vcf"
```

Comment:
do we really need to allow the pipeline to output uncompressed vcfs?

Reply:
I only look at uncompressed vcfs (the files are already tiny and it makes looking at things in Terra/Notebooks smoother). In fact I think I hardcoded `compress = false` when calling this task. Should I wire that argument through as an option?

Reply:
Ehhh it just irks me when I see uncompressed output (even if I know its tiny).  Tools you use in Terra/Notebooks should be able to read compressed vcfs :P.  I would lean towards wiring it through so people have an option to changing it.